### PR TITLE
Core 3484

### PR DESCRIPTION
**Pull request by dennisroberts71**

### CODE REVIEW

**Comment by dennisroberts71** [comment]
This PR includes changes for CORE-3484 and CORE-3485.

File: databases/de-database-schema/src/main/constraints/46_genome_ref.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+SET search_path = public, pg_catalog;
+
+--
+-- Uniqueness constraint on reference genome name.
+--
+ALTER TABLE ONLY genome_reference
+ADD CONSTRAINT genome_ref_mame_unique
```
Comment (ianmcorvidae) [inline databases/de-database-schema/src/main/constraints/46_genome_ref.sql]: It's just in the name of a constraint so it doesn't change functionality, but this says `mame`, not `name`

Comment (dennisroberts71) [inline databases/de-database-schema/src/main/constraints/46_genome_ref.sql]: ah. thanks for catching that

File: services/apps/src/apps/metadata/reference_genomes.clj [hunk 2]
```diff
@@ -83,18 +91,33 @@
     (sql/update genome_reference (set-fields update-values) (where {:id reference-genome-id}))
```
Comment (ianmcorvidae) [inline services/apps/src/apps/metadata/reference_genomes.clj]: I think that this function (`update-reference-genome`) should also validate that, if a new name or path is provided, it does not conflict with anything.

Comment (dennisroberts71) [inline services/apps/src/apps/metadata/reference_genomes.clj]: Good idea. I'll add that as well.

**Comment by ianmcorvidae** [comment]
Aside from the comment about `update-reference-genome` this looks straightforward, so as far as I'm concerned once that's in, :shipit:

**Comment by psarando** [comment]
Looks good :shipit:

**Pull request merged**